frontends/p4/entryPriorities.cpp

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2023 Mihai Budiu
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "entryPriorities.h"
+
+namespace P4 {
+
+const IR::Node *EntryPriorities::preorder(IR::EntriesList *entries) {
+    auto table = findContext<IR::P4Table>();
+    CHECK_NULL(table);
+    auto ep = table->properties->getProperty(IR::TableProperties::entriesPropertyName);
+    CHECK_NULL(ep);
+    if (ep->isConstant) return entries;
```

Comment:
I think that if method tableNeedsPriority() returns false for this table, that we should _not_ execute the code below that calculates entry priorities.  Instead, we should issue a warning or error if the source code contains explicit priority values for any entry.

If the P4 developer tries to specify priority values for such a table, I think they have a fundamental misunderstanding, and should get an error that says something like "This table cannot support entry priorities.  For example, tables with only exact and lpm match kinds cannot support entry priorities".

Reply:
Inside the zip file I have attached to this comment: https://github.com/p4lang/p4c/pull/3748#issuecomment-1556059443 the test program init-entries-error1-bmv2.p4 contains 3 different but similar table definitions that I believe should all give an error message at compile time for the reason mentioned in my comment above.